### PR TITLE
05875 query workflow overhaul

### PR DESCRIPTION
**Pull request by netopyr**
Some ideas to improve the query workflow:

- Extracted QueryMetrics
- Removed QueryContext and QueryContextImpl for now
- Removed SessionContext
- Added NodeInfo, NetworkInfo, CurrentPlatformStatus, and QueryFeeCheck with mono-adapters
- Turned FeeObject into a record and added totalFee()
- Some minor simplifications of QueryWorkflowImpl

### CODE REVIEW

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/ingest/IngestWorkflowImpl.java [hunk 1]
```diff
@@ -189,7 +186,7 @@ void submitTransaction(
         } catch (IOException ex) {
             // It may be that the response couldn't be written because the response buffer was
             // too small, which would be an internal server error.
-            throw new RuntimeException("Failed to write bytes to response buffer", ex);
+            throw new UncheckedIOException("Failed to write bytes to response buffer", ex);
```
Comment (rbair23) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/ingest/IngestWorkflowImpl.java:189]: Good one!

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/query/QueryMetrics.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.util.EnumMap;
+import java.util.Map;
+import javax.inject.Inject;
+
+/**
+ * A container for metrics related to queries.
+ */
+public class QueryMetrics {
```
Comment (rbair23) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/query/QueryMetrics.java:34]: Actually, I propose we eliminate these metrics entirely. They are already covered by the QueryMethod class in gRPC.

File: hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusGetTopicInfoHandler.java [hunk 3]
```diff
@@ -54,9 +54,17 @@
...
+
     @Inject
     public ConsensusGetTopicInfoHandler() {
-        // Exists for injection
+        // TODO: Not sure how to get the network info here.
+        this.networkInfo = null;
+    }
+
+    public ConsensusGetTopicInfoHandler(@NonNull final NetworkInfo networkInfo) {
+        this.networkInfo = requireNonNull(networkInfo);
```
Comment (rbair23) [inline hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusGetTopicInfoHandler.java:67]: So far this looks like the only thing we need to really dig into.

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/ingest/IngestWorkflow.java [hunk 4]
```diff
@@ -36,10 +35,8 @@ public interface IngestWorkflow {
...
      * unsuccessful calls). There are no unhandled exceptions (even Throwable is handled).
      *
-     * @param session The per-request {@link SessionContext}.
      * @param requestBuffer The raw protobuf transaction bytes. Must be a transaction object.
      * @param responseBuffer The raw protobuf response bytes.
      */
-    void submitTransaction(
-            @NonNull SessionContext session, @NonNull Bytes requestBuffer, @NonNull BufferedData responseBuffer);
+    void submitTransaction(@NonNull Bytes requestBuffer, @NonNull BufferedData responseBuffer);
 }
```
[the remark this answers is not in the document]
Comment (jasperpotts) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/ingest/IngestWorkflow.java:42]: Easy either way, happy for it to be removed and add back in later if we need it. We were thinking that there might be a bunch of resources that get used in a session like buffers that could get reused by reusing sessions after clearing them between queries. All a perf optimization that we can look at later if we need. Off-heap buffers are faster for parsing but are expensive to garbage collect so you want to pool and reuse them. Not sure if we need that extra bit of performance or not yet.

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/query/QueryChecker.java [hunk 5]
```diff
@@ -54,42 +60,61 @@ public class QueryChecker {
...
+     */
+    public void checkNodeState() throws PreCheckException {
+        if (nodeInfo.isSelfZeroStake()) {
+            // Zero stake nodes are currently not supported
+            throw new PreCheckException(INVALID_NODE_ACCOUNT);
+        }
+        if (currentPlatformStatus.get() != ACTIVE) {
+            throw new PreCheckException(PLATFORM_NOT_ACTIVE);
+        }
+    }
```
[the remark this answers is not in the document]
Comment (netopyr) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/query/QueryChecker.java:104]: Good idea! We should go through our list of general transaction checks to figure out what is also needed for queries.

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/query/QueryMetrics.java (hunk 2 above)
Comment (netopyr) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/query/QueryMetrics.java:34]: If both cases are handled (received and answered) I am fine with that.

Comment (netopyr) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/query/QueryMetrics.java]: Done

**Pull request closed without merging**